editors/vscode/src/common.ts

```diff
@@ -83,38 +94,128 @@ export class PropertiesViewProvider implements vscode.WebviewViewProvider {
+                        "",
+                        -1,
+                        -1,
+                        -1,
+                    );
+                    return;
+                }
+
+                const doc = editor.document;
+                const selection = editor?.selection;
+                const uri = doc.uri.toString();
+                const version = doc.version;
+                if (
+                    this.#current_uri === uri &&
+                    this.#current_version === version &&
+                    this.#current_cursor_line === selection.active.line &&
+                    this.#current_cursor_character ===
+                        selection.active.character
+                ) {
                     return;
                 }
-                let selection = event.selections[0];
-
-                query_properties(
-                    client,
-                    event.textEditor.document.uri,
-                    {
-                        line: selection.active.line,
-                        character: selection.active.character,
-                    },
-                    (p) => {
-                        const msg = {
-                            command: "set_properties",
-                            properties: p,
-                        };
-                        webviewView.webview.postMessage(msg);
-                    },
+                this.property_shown = this.update_view(
+                    doc.languageId,
+                    uri,
+                    version,
+                    selection.active.line,
+                    selection.active.character,
                 );
-                this.property_shown = true;
             },
         );
     }
 
+    refresh_view() {
+        this.update_view(
+            this.#current_language,
+            this.#current_uri,
+            this.#current_version,
+            this.#current_cursor_line,
+            this.#current_cursor_character,
+        );
+    }
+
+    private update_view(
+        language: string,
+        uri: string,
+        version: number,
+        line: number,
+        character: number,
+    ): boolean {
+        this.#current_uri = uri;
+        this.#current_version = version;
+        this.#current_cursor_line = line;
+        this.#current_cursor_character = character;
+        this.#current_language = language;
+
+        if (this._view === null) {
+            return false;
+        }
+        if (client === null) {
+            this._view?.webview.postMessage({
+                command: "show_error",
+                message: "Waiting for Slint LSP",
+            });
+            return false;
+        }
+        if (language !== "slint") {
+            if (this.property_shown) {
+                this._view?.webview.postMessage({
+                    command: "show_error",
+                    message:
+                        "Please open a <tt>.slint</tt> file to see properties",
```

Comment:
it should't be or look like an error. It should be an informational message.


![Screenshot_20221124_083634](https://user-images.githubusercontent.com/959326/203721506-6dc507a3-5fa1-4502-ac2b-e0ee1b7f51bf.png)

![Screenshot_20221124_083758](https://user-images.githubusercontent.com/959326/203721759-95460b03-4c66-4a8a-a84f-ac141cf48843.png)

![Screenshot_20221124_083900](https://user-images.githubusercontent.com/959326/203721945-0d653c21-483d-4d0a-8cb0-f351bcd369b7.png)


IMHO, the placeholder could be:

> The active editor is not displaying Slint file. 
>
> ![Slint logo simple](https://slint-ui.com/logo/slint-logo-simple-light.svg)

(the logo should be different in dark vs light style)



That's why i opted for `clear` instead of `show_error` because this is not an error condition, just normal operation.

I feel like it would benefit if the html for this was in propertiesView.ts, but that's not a strong feeling

Reply:
Looks much better than it does now! :+1: 

I think the massage could still mention Slint, as it is not obvious that this is a Slint widget otherwise.

Hnce the suggestion

> The active editor is not displaying a Slint file.